internal/worker/externalcontrollerupdater/externalcontrollerupdater.go

```diff
@@ -243,6 +244,13 @@ func (w *controllerWatcher) loop() error {
 			if err == w.catacomb.ErrDying() {
 				return err
 			} else if err != nil {
+				// If the controller cannot report its own addresses e.g. if it's
+				// behind a load-balancer then stop the worker without error.
+				if params.IsCodeNotSupported(err) {
+					logger.Debugf("external controller %q reported an unsupported error when attempting to watch for changes: %v", w.tag.Id(), err)
+					logger.Debugf("assuming controller cannot report its own addresses and completing watch for %q", w.tag.Id())
```

Comment:
Just do this in one line. No need to log the tag twice. And we know the error is "not supported" so no need to log it.